tests/instruments/quantum_machines/test_quantum_machines_cluster.py

```diff
@@ -214,7 +250,158 @@ def test_simulate(
 
         assert isinstance(job, MagicMock)
 
-    def test_set_parameter(self, qmm: QuantumMachinesCluster):
-        """Tests that set_parameter method raises a not implemented error."""
-        with pytest.raises(NotImplementedError, match="Setting a parameter is not supported for Quantum Machines yet."):
-            _ = qmm.set_parameter(parameter=Parameter.LO_FREQUENCY, value=0.0)
+    @pytest.mark.parametrize(
+        "bus, parameter, value",
+        [
+            ("drive_q0_rf", Parameter.LO_FREQUENCY, 6e9),
+            ("drive_q0_rf", Parameter.IF, 20e6),
+            ("drive_q0_rf", Parameter.GAIN, 0.001),
+            ("readout_q0_rf", Parameter.LO_FREQUENCY, 6e9),
+            ("readout_q0_rf", Parameter.IF, 20e6),
+            ("readout_q0_rf", Parameter.GAIN, 0.001),
+        ],
+    )
+    @patch("qililab.instruments.quantum_machines.quantum_machines_cluster.QuantumMachinesManager")
+    @patch("qililab.instruments.quantum_machines.quantum_machines_cluster.QuantumMachine")
+    def test_set_parameter_of_bus_method_with_octave(
+        self,
+        mock_qmm,
+        mock_qm,
+        bus: str,
+        parameter: Parameter,
+        value: float | str | bool,
+        qmm_with_octave: QuantumMachinesCluster,
+    ):
+        """Test the setup method with float value"""
+        qmm_with_octave.initial_setup()
+        qmm_with_octave.turn_on()
+        qmm_with_octave._config = qmm_with_octave.settings.to_qua_config()
+
+        qmm_with_octave.set_parameter_of_bus(bus, parameter, value)
+        if parameter == Parameter.LO_FREQUENCY:
+            qmm_with_octave._qm.octave.set_lo_frequency.assert_called_once()
+        if parameter == Parameter.GAIN:
+            qmm_with_octave._qm.octave.set_rf_output_gain.assert_called_once()
+        if parameter == Parameter.IF:
+            qmm_with_octave._qm.set_intermediate_frequency.assert_called_once()
+
+    @pytest.mark.parametrize(
+        "bus, parameter, value",
+        [
+            ("drive_q0", Parameter.IF, 20e6),
+        ],
```

Comment:
this you leave this in parametrize because you expect to add more parameters to be set by the non-octave buses in the future?

Reply:
Exactly. We will certainly add more parameters in the (near) future, but these 3 are the most important right now.